### DISCUSSION TITLE
Use different proxy for "Each Request" on the "Same Browser" instance

### DISCUSSION THREAD

**Discussion by SorceryAdmin** [cefsharp/CefSharp#3723] [Q&A]
I want to have the browser to use different proxies for each request when loading a page.

I tried to set proxy on the `OnBeforeResourceLoad` to one type and rever in the `OnResourceLoadComplete`
 basically is load, fonts, videos from a fast video streamming enabled one, and the rest of the content from the regular proxy or even not a proxy at all.
the issue on all my tests is that because the requests can be done in parallel the proxy set on the OnBeforeResourceLoad affect all the requests,

to set the proxy i use :

`   CefSharp.Cef.UIThreadTaskFactory.StartNew(delegate
                    {
                        var v = new Dictionary<string, object>
                        {
                            ["mode"] = "fixed_servers",
                            ["server"] = $"{proxy.Address.Scheme}://{proxy.Address.Host}:{proxy.Address.Port}"
                        };

                        if (!((ChromiumWebBrowser)chromiumWebBrowser).GetBrowser().GetHost().RequestContext.SetPreference("proxy", v, out error))
                            Console.WriteLine(error);
                    });`
is it possible at all to  do this or I'm  shooting for life on Mars?

**Comment by amaitland** [comment]
> is it possible at all to do this

No. That's not possible.
